src/v/coproc/router.cc

```diff
@@ -105,21 +139,26 @@ ss::future<> router::route() {
 }
 
 ss::future<> router::do_route() {
-    auto reducer =
-      [](std::vector<process_batch_request::data> acc, opt_req_data x) {
-          if (x.has_value()) {
-              acc.emplace_back(std::move(*x));
-          }
-          return acc;
-      };
     return ss::map_reduce(
              _sources.begin(),
              _sources.end(),
-             [this](std::pair<const model::ntp, topic_state>& p) {
-                 return route_ntp(p.first, p.second);
+             [this](router_source_manager::consumers_state::value_type& p) {
+                 return make_reader_cfg(p.second).then(
+                   [this, ntp = p.first, ts = p.second](
+                     opt_cfg config) mutable {
+                       if (!config) {
+                           return ss::make_ready_future<opt_req_data>();
+                       }
+                       return route_ntp(std::move(ntp), *config, ts);
+                   });
              },
              std::vector<process_batch_request::data>(),
-             std::move(reducer))
+             [](std::vector<process_batch_request::data> acc, opt_req_data x) {
+                 if (x.has_value()) {
+                     acc.emplace_back(std::move(*x));
+                 }
+                 return acc;
```

Comment:
does this make a copy ? should you take an r value ref ` && acc` and return a move (std::move(acc))

Reply:
No, this should be moved and elided.